rules/prefer-class-fields.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+const getIndentString = require('./utils/get-indent-string.js');
+
+const MESSAGE_ID = 'prefer-class-fields/error';
+const messages = {
+	[MESSAGE_ID]:
+		'Prefer class field declaration over `this` assignment in constructor for static values.',
+};
+
+/**
+ * @param {import('eslint').Rule.Node} node
+ * @returns {node is import('estree').ExpressionStatement & {expression: import('estree').AssignmentExpression & {left: import('estree').MemberExpression & {object: import('estree').ThisExpression}}}}
+ */
+const isThisAssignmentExpression = node => {
+	if (
+		node.type !== 'ExpressionStatement'
+		|| node.expression.type !== 'AssignmentExpression'
```

Comment:
The operator didn't checked

```js
			class Foo {
				constructor() {
					this.foo += 'foo';
				}
			}
```

Reply:
You're right, it should be ignored. Fixed here: 21e3cf1a172d1834bb67bbeabe39215cd0d3830e